### PR TITLE
fix: create interface for coverage plugin

### PR DESCRIPTION
**Pull request by minzcmu**
- create interface for coverage plugin
- it defines 2 required functions:
   - `getAccessToken()` => plugin should know how to get an access token to talk to the coverage server
   - `getUploadCoverageCmd()` => plugin should generate the shell commands to upload the coverage results to server

Related to: https://github.com/screwdriver-cd/screwdriver/issues/1007

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,6 +9,16 @@
 npm install screwdriver-screwdriver-coverage-base
 ```
 
+## Interface
+This is a interface for uploading code coverage results from a Screwdriver build to a code coverage server.
```
Comment (tkyi) [inline README.md]: `an` interface*

File: README.md [hunk 2]
```diff
@@ -9,6 +9,16 @@
...
+
+### getUploadCoverageCmd
+The `getUploadCoverageCmd` function should resolves a Promise with shell commands to upload code coverage results.
```
Comment (tkyi) [inline README.md]: should `resolve` a Promise

File: test/index.test.js [hunk 3]
```diff
@@ -3,7 +3,30 @@
...
+
+    afterEach(() => {
+        instance = null;
+    });
+
+    it('can create an coverage base class instance', () => {
+        assert.instanceOf(instance, CoverageBase);
+    });
+
+    it('should catch an error', () => {
```
Comment (tkyi) [inline test/index.test.js]: would be nice to nest under `describe('getUploadCoverageCmd',` and rename test to `returns not implemented` like https://github.com/screwdriver-cd/scm-base/blob/master/test/index.test.js#L96

**Review by tkyi — APPROVED** [review]
Minor comment, otherwise LGTM 👍

File: index.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    /**
+     * Get shell command to upload coverage to server
+     * @method getUploadCoverageCmd
+     * @return {Promise}     Shell commands to upload coverage
+     */
+    getUploadCoverageCmd() {
+        return this._uploadCoverage();
+    }
+
+    _uploadCoverage() {
```
Comment (tkyi) [inline index.js]: not sure if this name should match and be `_getUploadCoverageCmd`

Comment (minzcmu) [reply index.js]: loll it should, good catch!

**Review by tkyi — APPROVED** [review]
👍

File: README.md [hunk 5]
```diff
@@ -9,6 +9,27 @@
...
+This is an interface for uploading code coverage results from a Screwdriver build to a code coverage server.
+
+### getAccessToken
+##### Required Parameters
+| Parameter        | Type  |  Description |
+| :-------------   | :---- | :-------------|
+| buildCredentials        | Object | Information stored in the build JWT token |
+
+##### Expected Outcome
+The `getAccessToken` function should resolve a Promise with an access token that build can use to talk to the code coverage server.
```
Comment (d2lam) [inline README.md:22]: resolve to

**Review by d2lam — APPROVED** [review]
LGTM 👍

**Review by Zhongtang — APPROVED** [review]
👍

**Pull request merged**